src/emscripten-optimizer/optimizer-shared.cpp

```diff
@@ -67,7 +56,7 @@ AsmType detectType(Ref node, AsmData *asmData, bool inVarDef, IString minifiedFr
   switch (node[0]->getCString()[0]) {
     case 'n': {
       if (node[0] == NUM) {
-        if (!isInteger(node[1]->getNumber())) return ASM_DOUBLE;
+        if (!wasm::isInteger(node[1]->getNumber())) return ASM_DOUBLE;
```

Comment:
I was thinking of:
 * Applying the code as-is for now.
 * Factor out `support/` into a separate repo.
 * Factor our `emscripten_optimizer/` into a separate repo.
 * Use `support/` in the new `emscripten_optimizer` repo.
 * Update binaryen to use both as submodules.

We can dual-license `support` and `emscripten_optimizer`, first as MIT + Apache 2.0, and eventually as MIT + Apache 2.0 with exceptions.

Would that be OK?

Otherwise, we can do one of:
 1. Fork `emscripten_optimizer`.
 2. Make `emscripten_optimizer` configurable: use `wasm` bits only if present.